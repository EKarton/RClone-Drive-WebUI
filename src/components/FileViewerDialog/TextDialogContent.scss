.text-dialog-content {
  white-space: pre-wrap;
}
