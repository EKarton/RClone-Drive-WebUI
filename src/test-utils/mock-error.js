/**
 * Replaces the error object's stack trace with mock data
 * @param {Error} error the Error object
 * @returns {Error} the same error, but with the stack trace mocked
 */
export function mockErrorStackTrace(error) {
  error.stack = 'Mock content for stack trace';
  return error;
}
