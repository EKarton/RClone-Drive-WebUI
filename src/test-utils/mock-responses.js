export const mockFiles = {
  list: [
    {
      Path: '.rclone',
      Name: '.rclone',
      Size: -2,
      MimeType: 'inode/directory',
      ModTime: '2021-11-01T21:09:05.237Z',
      IsDir: true,
      ID: '1a4mI4vsGKypqFFPwLnrFid1ZdvlhirDi',
    },
    {
      Path: 'Documents',
      Name: 'Documents',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T01:51:05.982Z',
      IsDir: true,
      ID: '1_za5Kqy9ZjXZ7rsztGpKhtJLTuNCKv8q',
    },
    {
      Path: 'Games',
      Name: 'Games',
      Size: -9,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T05:14:59.237Z',
      IsDir: true,
      ID: '1j6mAjxQoJokiU8yWJYgyWYIiSIgH1G7M',
    },
    {
      Path: 'Literature',
      Name: 'Literature',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-29T07:10:05.059Z',
      IsDir: true,
      ID: '1dj1T5YkjRQZNiQ5efCA8PzideJb88d3K',
    },
    {
      Path: 'Pictures',
      Name: 'Pictures',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T03:39:11.850Z',
      IsDir: true,
      ID: '1D2PpmoIT23NqyTnolmohZ0UxvOd2D7n4',
    },
    {
      Path: 'Software',
      Name: 'Software',
      Size: -2,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T03:45:02.471Z',
      IsDir: true,
      ID: '1zzr0Bwp9tel66rCUKz8mZIXRBR2kuywr',
    },
    {
      Path: 'Videos',
      Name: 'Videos',
      Size: -8,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T04:22:24.997Z',
      IsDir: true,
      ID: '1eJlJ1PT7b9XY3Q5tLeU49BNkug259986',
    },
    {
      Path: 'backup.sh',
      Name: 'backup.sh',
      Size: 1110,
      MimeType: 'application/x-sh',
      ModTime: '2021-11-14T18:58:44.655Z',
      IsDir: false,
    },
  ],
};

export const mockRemotes = {
  remotes: ['googledrive', 'onedrive', 'icloud'],
};

export const mockPictures = {
  list: [
    {
      Path: 'Pictures/2012/20120517_171428.JPG',
      Name: '20120517_171428.JPG',
      Size: 335408,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T21:14:30.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_135206.JPG',
      Name: '20120517_135206.JPG',
      Size: 262719,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T17:52:06.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_204557.JPG',
      Name: '20120516_204557.JPG',
      Size: 245109,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T00:45:58.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_174510.JPG',
      Name: '20120517_174510.JPG',
      Size: 271914,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T21:45:12.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120820_144833.JPG',
      Name: '20120820_144833.JPG',
      Size: 2662880,
      MimeType: 'image/jpeg',
      ModTime: '2012-08-20T18:48:32.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_094409.JPG',
      Name: '20120517_094409.JPG',
      Size: 280596,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T13:44:10.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_181954.JPG',
      Name: '20120516_181954.JPG',
      Size: 416424,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-16T22:19:54.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_181809.JPG',
      Name: '20120516_181809.JPG',
      Size: 211575,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-16T22:18:10.000Z',
      IsDir: false,
    },
  ],
};
