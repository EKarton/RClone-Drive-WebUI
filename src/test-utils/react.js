import { RCloneInfoProvider } from 'contexts/RCloneInfo';
import { RecentPicturesProvider } from 'contexts/RecentPicturesList';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const customRender = (
  component,
  { initialRCloneInfoState, initialRecentPicturesState } = {},
  { route = '/' } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  const renderedComponent = render(
    <RCloneInfoProvider defaultState={initialRCloneInfoState}>
      <RecentPicturesProvider defaultState={initialRecentPicturesState}>
        <Router history={history}>{component}</Router>
      </RecentPicturesProvider>
    </RCloneInfoProvider>
  );

  return {
    ...renderedComponent,
    history,
  };
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent, render };
