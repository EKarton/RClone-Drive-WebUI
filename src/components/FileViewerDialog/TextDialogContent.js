import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import PropTypes from 'prop-types';
import { useState, useEffect } from 'react';
import './TextDialogContent.scss';

export default function TextDialogContent({ fileBlob }) {
  const [data, setData] = useState(null);

  useEffect(() => {
    fileBlob.text().then((contents) => setData(contents));
  }, [fileBlob]);

  return (
    <DialogContent>
      <DialogContentText className="text-dialog-content">{data}</DialogContentText>
    </DialogContent>
  );
}

TextDialogContent.propTypes = {
  fileBlob: PropTypes.instanceOf(Blob),
};
