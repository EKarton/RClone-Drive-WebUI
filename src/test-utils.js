import { FileViewerProvider, initialState } from 'contexts/FileViewerStore';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const customRender = (
  component,
  { initialFileViewerState = initialState } = {},
  { route = '/' } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  return render(
    <FileViewerProvider defaultState={initialFileViewerState}>
      <Router history={history}>{component}</Router>
    </FileViewerProvider>
  );
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent };
