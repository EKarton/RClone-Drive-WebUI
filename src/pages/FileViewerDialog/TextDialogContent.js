import { DialogContent, DialogContentText } from '@mui/material';
import { useState, useEffect } from 'react';
import './TextDialogContent.scss';

export default function TextDialogContent({ fileBlob }) {
  const [data, setData] = useState(null);

  useEffect(() => {
    fileBlob.text().then((contents) => setData(contents));
  }, [fileBlob]);

  return (
    <DialogContent>
      <DialogContentText className="text-dialog-content">{data}</DialogContentText>
    </DialogContent>
  );
}
