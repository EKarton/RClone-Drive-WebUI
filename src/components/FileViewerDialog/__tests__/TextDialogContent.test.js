import { Blob } from 'blob-polyfill';
import { customRender, waitFor } from 'test-utils/react';
import TextDialogContent from '../TextDialogContent';

describe('TextDialogContent', () => {
  it('should match snapshot', async () => {
    const blob = new Blob(['1234'], { type: 'application/text' });

    const component = customRender(<TextDialogContent fileBlob={blob} />);

    await waitFor(() => {
      expect(component.getByText('1234')).toBeInTheDocument();
      expect(component.baseElement).toMatchSnapshot();
    });
  });
});
