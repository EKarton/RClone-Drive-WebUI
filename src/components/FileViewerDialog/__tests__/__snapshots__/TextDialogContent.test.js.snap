// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextDialogContent should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
    >
      <p
        class="MuiDialogContentText-root MuiTypography-root MuiTypography-body1 text-dialog-content css-qfso29-MuiTypography-root-MuiDialogContentText-root"
      >
        1234
      </p>
    </div>
  </div>
</body>
`;
